Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -103,20 +112,65 @@ struct PackageController {
+                    .query(on: req.db,
+                           join: \Version.$package.$id == \Package.$id, method: .inner,
+                           join: \Package.$id == \Repository.$package.$id, method: .inner)
+                    .filter(Repository.self, \.$owner == owner) //TODO: Lowercase
+                    .filter(Repository.self, \.$name == repository)
+                    .filter(\Version.$docArchives != nil)
+                    .field(Version.self, \.$reference)
+                    .field(Version.self, \.$latest)
                     .field(Version.self, \.$packageName)
-                    .field(Version.self, \.$spiManifest)
+                    .field(Version.self, \.$docArchives)
                     .field(Repository.self, \.$ownerName)
-                    .first(),
-                      let body = awsResponse.body,
+                    .all()
+
+                let documentationVersions = queryResult.compactMap { result -> DocumentationVersion? in
+                    guard let reference = result.model.reference else { return nil }
+
+                    return DocumentationVersion(reference: reference,
+                                                ownerName: result.relation2?.ownerName ?? owner,
+                                                packageName: result.model.packageName ?? repository,
+                                                docArchives: (result.model.docArchives ?? []).map(\.title),
+                                                latest: result.model.latest)
+                }
+
+                guard let documentation = documentationVersions[reference: reference]
+                else {
+                    // If there's no match for this reference with a docArchive, we're done!
+                    let error = Abort(.notFound, reason: "No docArchives for this reference")
+                    return try await DocumentationErrorPage.View(path: req.url.path, error: error)
+                        .document()
+                        .encodeResponse(status: .notFound,
+                                        headers: req.headers.replacingOrAdding(name: .cacheControl, value: "no-cache"),
+                                        for: req)
+                }
+
+                let availableDocumentationVersions: [DocumentationPageProcessor.AvailableDocumentationVersion] = ([
+                    documentationVersions.filter { $0.latest == .defaultBranch }.first,
+                    documentationVersions.filter { $0.latest == .preRelease }.first
+                ] + documentationVersions.latestMajorVersions())
+                    .compactMap { version in
+                        guard let version = version,
+                              let latest = version.latest,
+                              let docArchives = version.docArchives
+                        else { return nil }
+
+                        return .init(kind: latest, reference: "\(version.reference)", docArchives: docArchives)
```

Comment:
```suggestion
                        guard let version = version,
                              let latest = version.latest
                        else { return nil }

                        return .init(kind: latest, reference: "\(version.reference)", docArchives: version.docArchives)
```

Reply:
Looks like that's also breaking the tests (compile error) - but how is that not failing for you locally? 🤔 

https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/runs/7395781849?check_suite_focus=true